core/src/main/java/kafka/server/SharePartition.java

```diff
@@ -32,208 +38,256 @@
+    public CompletableFuture<Optional<Throwable>> acknowledge(String memberId, List<AcknowledgementBatch> acknowledgementBatch) {
+        log.debug("Acknowledgement batch request for share partition: " + acknowledgementBatch);
+        throw new UnsupportedOperationException("Not implemented yet");
     }
 
-    public String toString() {
-      return "BatchRecord(" +
-          " baseOffset=" + baseOffset +
-          ", lastOffset=" + lastOffset +
-          ")";
-    }
+    static class BatchRecord implements Comparable<BatchRecord> {
+        private final long baseOffset;
+        private final long lastOffset;
 
-    @Override
-    public int compareTo(BatchRecord o) {
-        int res = Long.compare(this.baseOffset, o.baseOffset);
-        if (res != 0) {
-          return res;
+        public BatchRecord(long baseOffset, long lastOffset) {
+            this.baseOffset = baseOffset;
+            this.lastOffset = lastOffset;
         }
-        return Long.compare(this.lastOffset, o.lastOffset);
-    }
-  }
 
-  static class InFlightRecord {
-    private RecordState state;
-    private int deliveryCount;
+        public long baseOffset() {
+            return baseOffset;
+        }
 
-    public InFlightRecord(RecordState state) {
-      this(state, 0);
-    }
+        public long lastOffset() {
+            return lastOffset;
+        }
 
-    public InFlightRecord(RecordState state, int deliveryCount) {
-      this.state = state;
-      this.deliveryCount = deliveryCount;
-    }
+        @Override
+        public int hashCode() {
+            return Objects.hash(baseOffset, lastOffset);
+        }
+
+        @Override
+        public boolean equals(Object obj) {
+            if (this == obj) {
+                return true;
+            }
+            if (obj == null) {
+                return false;
+            }
+            if (getClass() != obj.getClass()) {
+                return false;
+            }
+            BatchRecord other = (BatchRecord) obj;
+            return baseOffset == other.baseOffset && lastOffset == other.lastOffset;
+        }
+
+        public String toString() {
+            return "BatchRecord(" +
+                " baseOffset=" + baseOffset +
+                ", lastOffset=" + lastOffset +
+                ")";
+        }
 
-    public RecordState messageState() {
-      return state;
+        @Override
+        public int compareTo(BatchRecord o) {
+            int res = Long.compare(this.baseOffset, o.baseOffset);
+            if (res != 0) {
+                return res;
+            }
+            return Long.compare(this.lastOffset, o.lastOffset);
+        }
     }
 
-    public void setMessageState(RecordState newState, boolean incrementDeliveryCount) {
-      state = newState;
-      if (incrementDeliveryCount) {
-        deliveryCount++;
+  static class InFlightFetchBatch {
+
+      // The member id of the client that is fetching the record.
+      private final String memberId;
```

Comment:
I was expecting Uuid. If we are generally using String, that's OK I guess, but a Uuid should be slightly more efficient for comparison and storage.

Reply:
Agree, just went for now with the memberid as defined in the request. Shall we change the fetch request as well?

Reply:
I am merging for now, I think we should change the request schema as well.